code/__HELPERS/icons.dm

```diff
@@ -897,3 +897,12 @@ proc/rand_hex_color()
 	for(var/i=0;i<6;i++)
 		color = color+pick(colors)
 	return "#[color]"
+
+//Imagine removing pixels from the main icon that are covered by pixels from the mask icon.
+proc/get_icon_difference(var/icon/main, var/icon/mask)
+	if(istype(main) && istype(mask))
+		mask.Blend(rgb(255,255,255), ICON_SUBTRACT) //Make all pixels on the mask as black as possible.
+		mask.Opaque(rgb(255,255,255)) //Make the transparent pixels (background) white.
+		mask.BecomeAlphaMask() //Make all the black pixels vanish (fully transparent), leaving only the white pixels.

```

Comment:
I never knew this was a thing in the iconprocs library and it could probably be used on the gibber now that I know about it.

Reply:
Yeah,  I really busted my brain trying to figure out how to write this proc and BecomeAlphaMask & the other icon helpers made me jump for joy. I think there's a miriad of places this one can be used.